### PR TITLE
Introduce an API field indicating whether safeguarding information has been supplied

### PR DESCRIPTION
**Pull request by AgaDufrat**
## Context

We do not want to send safeguarding information via the API as it means our clients become liable for looking after this sensitive data. They do need to know whether this information is present, though. Provider users will need to sign in to the web UI to see safeguarding information.

## Changes proposed in this pull request

Add new fields to the Application object:
- `safeguarding_issues_status`
- `safeguarding_issues_details_url`

## Guidance to review

- do the descriptions make sense?
- is the example URL (with id) ok?
- is it ok to include url_helpers?

## Link to Trello card

https://trello.com/c/4CO0fhJ7/2584-introduce-an-api-field-indicating-whether-safeguarding-information-has-been-supplied

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by duncanjbrown** [comment]
Is `safeguarding_issues_status` really a boolean? The enum in `ApplicationForm` has four options — I guess we'll never see `not_answered_yet`, but the other three are all still possible.

**Comment by AgaDufrat** [comment]
The A/C on the card said _"New field in API response: safeguarding_issues_status (true/false)"_ so I thought that's what was decided. Happy to change it to appropriate strings.

**Comment by duncanjbrown** [comment]
> > Is `safeguarding_issues_status` really a boolean? The enum in `ApplicationForm` has four options — I guess we'll never see `not_answered_yet`, but the other three are all still possible.
> 
> The A/C on the card said _"New field in API response: safeguarding_issues_status (true/false)"_ so I thought that's what was decided. Happy to change it to appropriate strings.

I think that was my bad, sorry. Let's do strings.

File: app/presenters/vendor_api/single_application_presenter.rb [hunk 1]
```diff
@@ -260,5 +264,9 @@ def work_history_breaks
         ''
       end
     end
+
+    def safeguarding_issues_details_url
+      application_form.has_safeguarding_issues_to_declare? ? provider_interface_application_choice_url(application_choice).concat('#criminal-convictions-and-professional-misconduct') : nil
```
Comment (duncanjbrown) [inline app/presenters/vendor_api/single_application_presenter.rb:269]: I think Rails path helpers support an `anchor:` property (https://stackoverflow.com/questions/51895357/rails-5-appending-an-html-anchor-to-rails-route-path-helper)

File: config/vendor-api-v1.yml [hunk 2]
```diff
@@ -471,6 +473,20 @@ components:
           description: Other personal or professional issues relevant to the application
             which are not covered in the form
           example: I am expecting to be called for jury service in the next 12 months
+        safeguarding_issues_status:
+          type: string
+          description: Status of candidate's response to the safeguarding issues declaration
+          enum:
+          - not_answered_yet
```
Comment (duncanjbrown) [inline config/vendor-api-v1.yml:480]: I don't think they'll ever get this?

**Review by duncanjbrown — APPROVED** [review]
Looks good — can we lose `not_answered_yet` as the form isn't complete (and therefore isn't sendable) without it?

File: config/vendor-api-v1.yml (hunk 2 above)
Comment (AgaDufrat) [reply config/vendor-api-v1.yml:480]: You are right, on providers side it can't be not_answered_yet.

**Pull request merged**